lib/AuthenticationSDK/util/JWT/JWTUtility.rb

```diff
@@ -0,0 +1,154 @@
+# frozen_string_literal: true
+
+require 'jwt'
+require 'json'
+require_relative 'JWTExceptions'
+
+module CyberSource
+  module Authentication
+    module Util
+      module JWT
+        # JWT Utility Class
+        #
+        # Provides JWT parsing, verification, and JWK handling functionality.
+        #
+        # @category Class
+        # @package  CyberSource::Authentication::Util::JWT
+        # @author   CyberSource
+        class JWTUtility
+          # Supported JWT algorithms
+          SUPPORTED_ALGORITHMS = %w[RS256 RS384 RS512].freeze
+
+          class << self
+            # Parses a JWT token and extracts its header, payload, and signature components
+            #
+            # @param jwt_token [String] The JWT token to parse
+            #
+            # @return [Hash] Hash containing header, payload, signature, and raw parts
+            # @raise [InvalidJwtException] If the JWT token is invalid or malformed
+            def parse(jwt_token)
+              raise InvalidJwtException.new('JWT token is null or undefined') if jwt_token.to_s.empty?
+
+              token_parts = jwt_token.split('.')
+              if token_parts.length != 3
+                raise InvalidJwtException.new('Invalid JWT token format: expected 3 parts separated by dots')
+              end
+
+              if token_parts.any?(&:empty?)
+                raise InvalidJwtException.new('Invalid JWT token: one or more parts are empty')
```

Comment:
Change this error message. It should not say that certain parts are missing. If some part is missing, it is a malformed JWT.

Say "Malformed JWT : JWT provided does not conform to the proper structure for JWT".